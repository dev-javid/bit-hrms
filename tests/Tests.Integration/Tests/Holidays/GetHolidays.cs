namespace Tests.Integration.Tests.Holidays
{
    public class GetHolidays(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/holidays?page=1&limit=10";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Holidays/GetHolidays.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
