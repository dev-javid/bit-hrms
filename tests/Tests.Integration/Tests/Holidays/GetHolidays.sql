INSERT INTO public.holidays (id, company_id, name, date, optional, created_by, created_on)
VALUES
(1, 999, 'Bank',         '2024-07-03', 'false', 1, now()),
(2, 999, 'Eid',          '2024-09-05', 'false', 1, now()),
(3, 999, 'Holi',         '2024-10-09', 'false', 1, now()),
(4, 999, 'Diwali',       '2024-05-11', 'false', 1, now()),
(5, 999, 'Ramadan',      '2024-11-23', 'false', 1, now()),
(6, 999, 'Republic Day', '2024-05-15', 'false', 1, now()),
(7, 999, 'IT',           '2024-12-18', 'false', 1, now());
