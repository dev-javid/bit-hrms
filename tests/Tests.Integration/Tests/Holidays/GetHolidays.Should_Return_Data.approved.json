{
    "page": 1,
    "limit": 10,
    "total": 7,
    "items": [
        {
            "holidayId": 4,
            "date": "2024-05-11",
            "name": "Diwali",
            "optional": false
        },
        {
            "holidayId": 6,
            "date": "2024-05-15",
            "name": "Republic Day",
            "optional": false
        },
        {
            "holidayId": 1,
            "date": "2024-07-03",
            "name": "Bank",
            "optional": false
        },
        {
            "holidayId": 2,
            "date": "2024-09-05",
            "name": "Eid",
            "optional": false
        },
        {
            "holidayId": 3,
            "date": "2024-10-09",
            "name": "Holi",
            "optional": false
        },
        {
            "holidayId": 5,
            "date": "2024-11-23",
            "name": "Ramadan",
            "optional": false
        },
        {
            "holidayId": 7,
            "date": "2024-12-18",
            "name": "IT",
            "optional": false
        }
    ]
}